.github/workflows/publish-release.yml

```diff
@@ -40,13 +40,12 @@ jobs:
           restore-keys: ${{ runner.os }}-m2
       - name: 🤳 Build Native Quarkus
         run: mvn install -Pnative -DskipTests
-      - name: 🛂 Find the version
-        run: 'echo "version=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)" >> $GITHUB_ENV'
       - name: Upload binaries to release
         uses: svenstaro/upload-release-action@v2
         with:
           repo_token: ${{ secrets.GITHUB_TOKEN }}
-          file: api/target/api-${{ env.version }}${{ matrix.file }}
+          file: api/target/api-*${{ matrix.file }}
+          file_glob: true
           asset_name: ${{ matrix.asset_name }}
```

Comment:
rereading the doc, in this case, the asset_name is ignored `file_glob: If set to true, the file argument can be a glob pattern (asset_name is ignored in this case) (Default: false)`

Reply:
an alternative could be to configure the output name of quarkus:
https://access.redhat.com/documentation/en-us/red_hat_build_of_quarkus/1.3/html/developing_and_compiling_your_quarkus_applications_with_apache_maven/proc-configuring-the-project-output_quarkus-maven
`quarkus.package.output-name`